ooipy/hydrophone/basic.py

```diff
@@ -250,8 +251,20 @@ def compute_spectrogram(
             self.spectrogram = None
             return None
         else:
-            self.spectrogram = Spectrogram(np.array(time), np.array(f), np.array(specgram))
-            return self.spectrogram
+            spec_xr = xr.DataArray(
+                np.array(specgram),
+                dims=["time", "frequency"],
+                coords={"time": np.array(time), "frequency": np.array(f)},
+                attrs=dict(
+                    start_time=self.stats.starttime.datetime,
+                    end_time=self.stats.endtime.datetime,
+                    nperseg=L,
+                    units="dB rel µ Pa^2 / Hz",
+                ),
+                name="spectrogram",
+            )
+            # self.spectrogram = Spectrogram(np.array(time), np.array(f), np.array(specgram))
```

Comment:
I think we can uncomment line 266 and just save spec_xr to self spectrogram

```
self.spectrogram = spec_xr
```

Reply:
Actually, I'm thinking that this feature of saving the spectrograms and psd as an attribute to the HydrophoneData object might not be that useful, so we can instead just remove these lines instead of commenting them out.

Can you check to make sure that there is no mention of saving the specs or psds in the HydrophoneData object in any of the doc strings?

Reply:
The doc strings for class Hydrophone Data are as follows:

```
"""
    Object that stores hydrophone data

    Attributes
    ----------
    spectrogram : Spectrogram
        spectrogram of HydrophoneData.data.data.
        Spectral level, time, and frequency bins can be accessed by
        spectrogram.values, spectrogram.time, and spectrogram.freq
    psd : Psd
        power spectral density estimate of HydrophoneData.data.data.
        Spectral level and frequency bins can be accessed by
        psd.values and psd.freq
    psd_list : list of :class:`.Psd`
        the data object is divided into N segments and for each
        segment a separate power spectral density estimate is computed and
        stored in psd_list. psd_list is computed by compute_psd_welch_mp
    type : str
        Either 'broadband' or 'low_frequency' specifies the type of hydrophone
        that the date is from.

    """
```

It has these attributes mentioned, so would that mean we need to save those? @John-Ragland

Reply:
Let's edit the doc string and remove the spectrogram, psd, and psd_list attributes. And then while we're at it, let's add a section for the methods:
```
Methods
---------
frequency_calibration
compute_spectrogram
compute_spectrogram_mp
compute_psd_welch
wav_write
get_asset_ID
save
```

I think there might be a way to make these methods automatically link to their section in the docs, but I don't know how to do that.

Reply:
Okay edited the doc string accordingly. Also changed the doc strings for the individual functions to include the new return type of the objects.